crates/bevy_debug_draw/src/debug_draw.rs

```diff
@@ -0,0 +1,200 @@
+use std::f32::consts::TAU;
+
+use bevy_asset::Handle;
+use bevy_ecs::system::Resource;
+use bevy_math::{vec3, Quat, Vec2, Vec3};
+use bevy_render::prelude::{Color, Mesh};
+
+/// A resource with an immediate mode API for drawing lines and wireshapes.
+/// Useful for visual debugging.
+#[derive(Resource)]
+pub struct DebugDraw {
+    positions: Vec<[f32; 3]>,
+    colors: Vec<[f32; 4]>,
+    pub(crate) mesh_handle: Option<Handle<Mesh>>,
+    /// The amount of line segments to use when drawing a circle.
+    ///
+    /// Defaults to `24`.
+    pub circle_segments: u32,
+}
+
+impl Default for DebugDraw {
+    fn default() -> Self {
+        DebugDraw {
+            positions: Vec::new(),
+            colors: Vec::new(),
+            mesh_handle: None,
+            circle_segments: 24,
+        }
+    }
+}
+
+impl DebugDraw {
```

Comment:
I did implement a collection of outlines here if you want a reference. Though unlike this, which uses `PrimitiveTopology::LineList`, it assumes a `PrimitiveTopology::LineStrip`, which divide by two the number of required indices if your lines are continuous. <https://github.com/nicopap/bevy-cool-shapes/blob/d84c8efcb4bbccd188a1ca8752345b44a7812a7c/bevy_cool_shapes_render/src/outline.rs>

Reply:
Totally forgot that it's possible to use NaNs to split a LineStrip!